### PR TITLE
Updated dependencies for openforcefield toolkit.

### PR DESCRIPTION
**Pull request by andrrizzi**
Fix some of the dependencies for the conda dev package. I've also added the `extra.upload: main` to fix the label of the anaconda package.

I've mostly taken the dependencies from [this conda recipe](https://github.com/openforcefield/openforcefield/blob/84487d9ced489d9d354c00459c16931f3abbb68e/devtools/conda-recipe/meta.yaml) from the `openforcefield/openforcefield` repo.

The main question is: should we remove some of the optional packages? Installing a fresh conda environment using [this recipe](https://github.com/omnia-md/conda-recipes/blob/2de9a1fcd9f8731a58d99422c0d1d0f9f1c25dcf/openforcefield/meta.yaml) on `omnia-md/conda-recipes` took several minutes with the CPU under a heavy workload with my MacBook Pro so if we're planning to have a hands-on session in which people would like to install everything we may need to strip the recipe down to a bare minimum.

### CODE REVIEW

File: openforcefield/meta.yaml [hunk 1]
```diff
@@ -25,9 +26,19 @@ requirements:
     - networkx >=2.0
     - lxml
     - icu 58*  # This is a lxml dependency but sometimes conda installs version 56
+    - packaging
+    # Should these be optional?
     - rdkit
     - ambertools
     #- parmed # uncomment when new ambermini (without parmed) is built
+    # Should we require this? It's useful for tests and examples, but not strictly required
+    - openmmtools # If we remove this we need to install openmm (which minimum version?)
```
Comment (jchodera) [inline openforcefield/meta.yaml]: Are we using `openmmtools`? Maybe we should just install `openmm` instead?

Comment (jchodera) [reply openforcefield/meta.yaml]: I don't think we use `openmmtools`. 
image
Let's just change this to `openmm`.

Comment (andrrizzi) [reply openforcefield/meta.yaml]: Actually no, I just copied from the recipe in the source repo, although there's a weird reference to `openmmtools.topology` in the `SMIRNOFF_comparison` scripts.

I'll substitute it with `openmm` without minimum version for now.

**Comment by andrrizzi** [comment]
Merging for now to test this build. We can always remove the optional dependencies later if the installation is too slow.

**Comment by j-wags** [comment]
Yes -- I agree with switching down to `openmm` from `openmmtools` to improve the build time for the workshop. The `SMIRNOFF_comparison` example is important, but, even if losing openmmtools breaks it (which I haven't looked into yet) we can bring it back online later. Getting the `SMIRNOFF_comparison` example back online is actually fairly high on my to-do list, so I think I'll run into any real breakages there quickly.

**Pull request merged**